File: ruby/emitter.rb
require 'bunny'

# Set up the connection to the RabbitMQ server
connection = Bunny.new(:username => "user", :password => "password")
connection.start

# Set up the channel and queue in RabbitMQ
channel = connection.create_channel
queue = channel.queue('hello')

# Start reading user input
while true do
    print "> "
    message = gets.strip! # Remove newline at end of message
    # Send the message to RabbitMQ
    channel.default_exchange.publish(message, routing_key: queue.name)
    puts "Published \"#{message}\" on #{queue.name}"
end

File: node/emitter.js
#!/usr/bin/env node

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});
var amqp = require('amqplib/callback_api');

// Recursive function to read user input for each message to send
var readInput = function(channel, queue) {
    readline.question('> ', (msg) => {
        // Send message to RabbitMQ
        channel.sendToQueue(queue, Buffer.from(msg));
        console.log("Message Sent: %s", msg);
        readInput(channel, queue);
    });
}

// Connect to the rabbitmq server, channel, and queue
amqp.connect('amqp://localhost', function(error0, connection) {
  connection.createChannel(function(_, channel) {
    var queue = 'hello';
    channel.assertQueue(queue, {
      durable: false
    });

    // Begin reading user input
    readInput(channel, queue);
  });
});